upcloud_api/server.py

```diff
@@ -505,28 +504,35 @@ def get_ip(self, access='public', addr_family=None, strict=None):
         if addr_family not in ['IPv4', 'IPv6', None]:
             raise Exception("`addr_family` must be 'IPv4', 'IPv6' or None")
 
-        if access not in ['private', 'public']:
-            raise Exception("`access` must be 'public' or 'private'")
+        if access not in ['private', 'public', 'utility']:
+            raise Exception("`access` must be 'public', 'utility' or 'private'")
 
-        if not hasattr(self, 'ip_addresses'):
-            self.populate()
+        if not hasattr(self, 'networking'):
+            raise Exception(
+                "`networking` attribute is missing, server details must be fetched first"
+            )
 
-        # server can have several public or private IPs
-        ip_addrs = [ip_addr for ip_addr in self.ip_addresses if ip_addr.access == access]
+        ip_addrs = []
+        for iface in self.networking['interfaces']['interface']:
+            ip_addresses = iface['ip_addresses']['ip_address']
```

Comment:
`ip_addrs` and `ip_addresses` in same scope feels a bit funny – what about `iface_ip_addrs` here? 🤔 

```suggestion
            iface_ip_addrs = iface['ip_addresses']['ip_address']
```

Reply:
Sure!

Reply:
Changed.